### PR TITLE
Goa 2313 annotation add date to indexing and model

### PR DESCRIPTION
**Pull request by eddturner**

### CODE REVIEW

File: annotation-common/src/main/java/uk/ac/ebi/quickgo/annotation/common/document/AnnotationFields.java [hunk 1]
```diff
@@ -35,6 +35,7 @@
     public final static String REFERENCE_SEARCH = "referenceSearch";
     public final static String WITH_FROM_SEARCH = "withFromSearch";
     public final static String GO_ASPECT = "goAspect";
+    public final static String DATE = "date";
```
Comment (rantunesboreas) [inline annotation-common/src/main/java/uk/ac/ebi/quickgo/annotation/common/document/AnnotationFields.java]: Could you add this to the retrievable fields. I know its empty now, but I will be changing that soonish.

File: indexing/src/test/java/uk/ac/ebi/quickgo/index/annotation/AnnotationValidatorTest.java [hunk 2]
```diff
@@ -279,19 +279,24 @@ public void invalidSingleTermedComponentAnnotationProperty() {
 
     @Test(expected = ValidationException.class)
     public void invalidAnnotationPropertyDueToMissingGOEvidence() {
-        annotation.annotationProperties = "taxon_id=35758|db_subset=TrEMBL|db_object_symbol=moeA5|db_object_type=protein";;
+        annotation.annotationProperties =
+                "taxon_id=35758|db_subset=TrEMBL|db_object_symbol=moeA5|db_object_type=protein";
+        ;
```
Comment (rantunesboreas) [inline indexing/src/test/java/uk/ac/ebi/quickgo/index/annotation/AnnotationValidatorTest.java]: Extra semi-colon, please remove.

File: indexing/src/test/java/uk/ac/ebi/quickgo/index/annotation/AnnotationValidatorTest.java [hunk 3]
```diff
@@ -279,19 +279,24 @@ public void invalidSingleTermedComponentAnnotationProperty() {
...
         validator.validate(annotation);
     }
 
     @Test(expected = ValidationException.class)
     public void invalidAnnotationPropertyDueToMissingTaxonId() {
-        annotation.annotationProperties = "go_evidence=IEA|db_subset=TrEMBL|db_object_symbol=moeA5|db_object_type=protein";;
+        annotation.annotationProperties =
+                "go_evidence=IEA|db_subset=TrEMBL|db_object_symbol=moeA5|db_object_type=protein";
+        ;
```
Comment (rantunesboreas) [inline indexing/src/test/java/uk/ac/ebi/quickgo/index/annotation/AnnotationValidatorTest.java]: Extra semi-colon please remove.

File: indexing/src/test/java/uk/ac/ebi/quickgo/index/annotation/AnnotationValidatorTest.java [hunk 4]
```diff
@@ -279,19 +279,24 @@ public void invalidSingleTermedComponentAnnotationProperty() {
...
         validator.validate(annotation);
     }
 
     @Test(expected = ValidationException.class)
     public void invalidAnnotationPropertyDueToMissingDbObjectType() {
-        annotation.annotationProperties = "go_evidence=IEA|taxon_id=35758|db_subset=TrEMBL|db_object_symbol=moeA5";;
+        annotation.annotationProperties = "go_evidence=IEA|taxon_id=35758|db_subset=TrEMBL|db_object_symbol=moeA5";
+        ;
```
Comment (rantunesboreas) [inline indexing/src/test/java/uk/ac/ebi/quickgo/index/annotation/AnnotationValidatorTest.java]: Extra semi-colon please remove.

File: annotation-common/src/main/java/uk/ac/ebi/quickgo/annotation/common/document/AnnotationFields.java (hunk 1 above)
Comment (eddturner) [inline annotation-common/src/main/java/uk/ac/ebi/quickgo/annotation/common/document/AnnotationFields.java]: done

File: indexing/src/test/java/uk/ac/ebi/quickgo/index/annotation/AnnotationValidatorTest.java (hunk 2 above)
Comment (eddturner) [inline indexing/src/test/java/uk/ac/ebi/quickgo/index/annotation/AnnotationValidatorTest.java]: done

File: indexing/src/test/java/uk/ac/ebi/quickgo/index/annotation/AnnotationValidatorTest.java (hunk 3 above)
Comment (eddturner) [inline indexing/src/test/java/uk/ac/ebi/quickgo/index/annotation/AnnotationValidatorTest.java]: done

**Pull request merged**